packages/shopify-codemod/test/fixtures/coffeescript-soak-to-condition/assignment.output.js

```diff
@@ -0,0 +1,16 @@
+if (typeof bar === 'function' && bar() != null && bar().baz.qux != null) {
```

Comment:
I've spotted a couple of places where this is a performance concern (parsing JSON responses, triggering reflows).  My PR to simplify some CoffeeScript files will take care of this.